### DISCUSSION TITLE
Is it possible to pass a delegate to a method called via MSBuild, and if so how?

### DISCUSSION THREAD

**Discussion by IanKemp** [dotnet/msbuild#10109] [Q&A] [answered]
I have the following C# code to convert a kebab case string (all lowercase, words separated via dashes) to camel case (first character of every word starts with uppercase), which I do with the following code that calls `Regex.Replace(string, string, MatchEvaluator)`:

```csharp
string input = "kebab-case";

var camelCased = Regex.Replace(input, "-.", m => m.Value.ToUpper().Substring(1)); // passing an anonymous delegate

Console.WriteLine(camelCased); // "kebabCase"
```

I need to convert this code to a format that MSBuild understands so that I can execute it in an `AfterBuild` target, but I can't figure out the correct syntax to use. I was under the impression that MSBuild's syntax is identical to PowerShell and in the latter this would be expressed as:

```powershell
[System.Text.RegularExpressions.Regex]::Replace($input, "-.", {  param($m) $m.Value.ToUpper().Substring(1) })
```

but the above doesn't work in MSBuild (it treats the delegate as a literal string), nor does any other variation I've tried (outputs "The expression ... cannot be evaluated"). Is this something that MSBuild even supports and if so, what's the correct syntax to use?

**Comment by rainersigwald** [comment, marked as answer]
No, MSBuild doesn't support any mechanism to express a method/delegate/lambda in a property function.

You should probably consider putting this in an [inline task](https://learn.microsoft.com/visualstudio/msbuild/msbuild-roslyncodetaskfactory).
